web/hooks/use-post-supabase.ts

```diff
@@ -0,0 +1,57 @@
+import { Group } from 'common/group'
+import { Post } from 'common/post'
+import { useEffect, useState } from 'react'
+import { db } from 'web/lib/supabase/db'
+import { getPost } from 'web/lib/supabase/post'
+
+export function useRealtimePost(postId?: string) {
+  const [post, setPost] = useState<Post | null>(null)
+  function fetchPost() {
+    if (postId) {
+      getPost(postId)
+        .then((result) => {
+          setPost(result)
+        })
+        .catch((e) => console.log(e))
+    }
+  }
+
+  useEffect(() => {
+    fetchPost()
+  }, [postId])
+
+  useEffect(() => {
+    if (postId) {
+      const channel = db.channel('post-realtime')
+      channel.on(
+        'postgres_changes',
+        {
+          event: '*',
+          schema: 'public',
+          table: 'posts',
+          filter: `id=eq.${postId}`,
+        },
+        (payload) => {
```

Comment:
I'm pretty sure payload contains the updated post... Did you console.log it?
Then you would not need to fetch it at all.